Client/Frontend/Settings/SettingsTableViewController.swift

```diff
@@ -585,3 +602,48 @@ class SettingsTableViewController: UITableViewController {
         return nil
     }
 }
+
+private class SettingsTableSectionHeaderView: UIView {
+    lazy var titleLabel: UILabel = {
+        var headerLabel = UILabel()
+        var frame = headerLabel.frame
+        frame.origin.x = 15
+        frame.origin.y = 17
+        headerLabel.frame = frame
+        headerLabel.textColor = AppConstants.TableViewHeaderTextColor
+        headerLabel.font = AppConstants.DefaultStandardFontBold
+        return headerLabel
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.backgroundColor = AppConstants.TableViewHeaderBackgroundColor
+    }
+
+    required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private override func drawRect(rect: CGRect) {
+        var topBorder = CALayer()
+        topBorder.frame = CGRectMake(0.0, 0.0, rect.size.width, 0.5)
+        topBorder.backgroundColor = AppConstants.SeparatorColor.CGColor
+        layer.addSublayer(topBorder)
+        var bottomBorder = CALayer()
+        bottomBorder.frame = CGRectMake(0.0, rect.size.height - 0.5 , rect.size.width, 0.5)
+        bottomBorder.backgroundColor = AppConstants.SeparatorColor.CGColor
+        layer.addSublayer(bottomBorder)
+    }
+
+    private override func layoutSubviews() {
+        if let label = titleLabel.text {
+            if titleLabel.superview == nil {
+                addSubview(titleLabel)
+            }
+            titleLabel.sizeToFit()
+        }
+        else {
+            titleLabel.removeFromSuperview()
```

Comment:
It's basically an optimisation. If the header has no value then we don't include a label and the view has 1 less thing to render.  Although this works better with header reuse rather than with 1 time header creation and I realised that I am not using header reuse.

However, on reflection, this is a bit of a swift/obj-c confusion. In Obj-C you would check to see if the underlying var `_titleLabel` exists. By referencing `_titleLabel` rather than `self.titleLabel` you check for the existence of the var without invoking the lazy initialiser. Therefore you can add the subview if and only if it has been created by being reference outside of the view (using `titleLabel.text`). This is a pattern that I habitually used a _lot_ in Obj-C ensuring that no more views that are required are created or added. 

In  Swift I am unsure how to achieve the same effect. By referencing `titleLabel` to check of it's existence you create it and therefore defeat the purpose of using a lazy initialiser here.

I shall update to extend `UITableViewHeaderFooterView` and use section header reuse in `viewForHeaderInSection` which is better and remove the lazy init, just calling `titleLabel.sizeToFit` in `layoutSubviews`